src/game/common/ini.cpp

```diff
@@ -400,379 +405,452 @@ int INIClass::Enumerate_Entries(const char *section, const char *entry_prefix, u
+        strlcpy(buffer, block_ptr, sizeof(buffer));
+        snprintf(entry, sizeof(entry), "%d", line);
 
-            if (block_len <= 0) {
-                break;
-            }
+        block_len = strlen(buffer);
 
-            if (block_len >= MAX_TEXTBLOCK_LINE_LENGTH) {
-                while (!isspace(buffer[block_len])) {
-                    if (!--block_len) {
-                        return true;
-                    }
-                }
+        if (block_len <= 0) {
+            break;
+        }
 
-                if (block_len <= 0) {
-                    break;
+        if (block_len >= MAX_TEXTBLOCK_LINE_LENGTH) {
+            while (!isspace(buffer[block_len])) {
+                if (!--block_len) {
+                    return true;
                 }
+            }
 
-                buffer[block_len] = '\0';
+            if (block_len <= 0) {
+                break;
             }
 
-            strtrim(buffer);
-            Put_String(section, entry, buffer);
-            ++line;
+            buffer[block_len] = '\0';
         }
 
-        return true;
+        strtrim(buffer);
+
+        Put_String(section, entry, buffer);
+
+        ++line;
     }
 
-    return false;
+    return true;
 }
 
 int INIClass::Get_TextBlock(const char *section, char *buffer, int length) const
 {
     captainslog_assert(buffer != nullptr);
     captainslog_assert(length != 0);
 
+    if (section == nullptr || buffer == nullptr || length <= 0) {
+        return 0;
+    }
+
     int total = 0;
 
-    if (section != nullptr && buffer != nullptr && length > 0) {
-        // If buffer has a length, we have at least a null string
-        buffer[0] = '\0';
+    // If buffer has a length, we have at least a null string
+    buffer[0] = '\0';
 
-        int elen = Entry_Count(section);
+    int elen = Entry_Count(section);
 
-        // If buffer can fit at least one char as well, get a char.
-        for (int i = 0; i < elen && length > 1; ++i) {
-            if (i > 0) {
-                // Puts a space between lines
-                *buffer++ = ' ';
-                --length;
-                ++total;
-            }
+    // If buffer can fit at least one char as well, get a char.
+    for (int i = 0; i < elen && length > 1; ++i) {
+        if (i > 0) {
+            // Puts a space between lines
+            *buffer++ = ' ';
+            --length;
+            ++total;
+        }
 
-            Get_String(section, Get_Entry(section, i),"", buffer, length);
+        Get_String(section, Get_Entry(section, i),"", buffer, length);
 
-            total = strlen(buffer);
-            length -= total;
-            buffer += total;
-        }
+        total = strlen(buffer);
+        length -= total;
+        buffer += total;
     }
 
     return total;
 }
 
 BOOL INIClass::Put_Int(const char *section, const char *entry, int value, int format)
 {
-    char buffer[512];
+    captainslog_assert(section != nullptr);
```

Comment:
This function does nothing with these values, it would be up to Put_String to assert or handle them.